test/contrib/test_socks.py

```diff
@@ -229,6 +229,37 @@ def request_handler(listener):
         self.assertEqual(response.data, b'')
         self.assertEqual(response.headers['Server'], 'SocksTestServer')
 
+    def test_local_dns(self):
+        def request_handler(listener):
+            sock = listener.accept()[0]
+
+            handler = handle_socks5_negotiation(sock, negotiate=False)
+            addr, port = next(handler)
+
+            self.assertIn(addr, ['127.0.0.1', '::1'])
```

Comment:
You will need to start using `unittest2` in these test cases to use `assertIn`: it's not present on Python 2.6.